### PR TITLE
IRGen: gatherbits - extend the bits before we shift left

### PR DESCRIPTION
**Pull request by aschwaighofer**
SR-9335
rdar://46264497

### CODE REVIEW

**Comment by aschwaighofer** [comment]
@swift-ci Please smoke test

File: test/Interpreter/multi_payload_shifting.swift [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// RUN: %empty-directory(%t)
+
+// RUN: %target-build-swift -o %t/a.out %s
+// RUN: %target-run %t/a.out 2>&1
```
Comment (jrose-apple) [inline test/Interpreter/multi_payload_shifting.swift]: Nitpick: This should use `%target-run-simple-swift`, which is a slightly more flexible substitution. There's also no need to pipe `2>&1`.

**Comment by aschwaighofer** [comment]
@swift-ci Please test

File: test/Interpreter/multi_payload_shifting.swift [hunk 2]
```diff
@@ -1,7 +1,6 @@
 // RUN: %empty-directory(%t)
 
-// RUN: %target-build-swift -o %t/a.out %s
-// RUN: %target-run %t/a.out 2>&1
+// RUN: %target-run-simple-swift %s
```
Comment (jrose-apple) [inline test/Interpreter/multi_payload_shifting.swift]: Ah, sorry, this substitution has the `%s` in it already. You also don't need the `%empty-directory` in this case.

**Comment by aschwaighofer** [comment]
@swift-ci Please smoke test

**Pull request merged**